teps/0154-concise-remote-resolver-syntax.md

```diff
@@ -0,0 +1,336 @@
+
+Examples:
+```
+https://github.com/wlynch/test@main#task.yaml
+git://github.com/wlynch/test@main#task.yaml
+oci://wlynch/image:v1@sha256:...#task/foo?secret=my-secret
+https://example.com/task.yaml
+http://10.5.3.2:8080/task.yaml
+artifact/tasks-catalog/tasks/task@v1.0 # e.g. for the hub resolver where there is no need for a scheme
+```
+
+#### Git Resolver
+Based on `git` resolver's [params](https://tekton.dev/docs/pipelines/git-resolver/#parameters), we define the following `resolver-reference` format:
+
+```
+<scheme>://<complete git repo>@<revision>#<pathInRepo>?token=<token>&tokenKey=<tokenKey>&serverURL=<serverURL>&scmType=<scmType>
+```
+
+Example:
+
+```
+https://github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token
+git://github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token
+git+https://github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token
+```
+
+Using the above example, a user can do the following when referencing the Task:
+```yaml
+taskRef:
+  resolver: git
+  url: https://github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token
+```
+
+#### Bundle Resolver
+Based on `bundle` resolver's [params](https://tekton.dev/docs/pipelines/bundle-resolver/#parameters), we define the following `resolver-reference` format: 
+
+```
+oci://<bundle-provider>/<bundle-namespace>/<bundle-name>@<bundle-version>?kind=<kind>&name=<name>&secret=<secret>
+```
+
+Example
+
+```
+oci://docker.io/ptasci67/example-oci@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828?kind=task&name=hello-world?secret=default
+```
+
+Using the above example, a user can do the following when referencing the Task:
+```yaml
+taskRef:
+  resolver: bundles
+  url: "oci://docker.io/ptasci67/example-oci@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828?kind=task&name=hello-world?secret=default"
+```
+
+Occasionally, users may want to fetch from a private registry (e.g. at `10.96.190.208:5000`).
+
+```yaml
+taskRef:
+  resolver: bundles
+  url: "oci://10.96.190.208:5000/ptasci67/example-oci@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828?kind=task&name=hello-world?secret=default"
+```
+
+#### Http Resolver
+Based on `http` resolver's [params](https://tekton.dev/docs/pipelines/http-resolver/#parameters), we define the following `resolver-reference` format: 
+
+```
+<http-url>?http-username=<http-username>&http-password-secret=<http-password-secret>&http-password-secret-key=<http-password-secret-key>
```

Comment:
isn't this too specific to k8s secrets?

Reply:
Those are the currently chosen parameter names for the http resolver:  https://tekton.dev/docs/pipelines/http-resolver/#parameters 

I just used them here for parity. We can probably change them since it is still alpha. But I think it should be consistent and is probably outside the scope of this TEP? (as in we update the names if desired separately)

Reply:
@dibyom isn't that what we want ? being able to pass secrets and key location ? most tekton tasks (in catalog) have that same sort of pattern..

@chitrangpatel all good by me, just wondering how would you handle non url parseable characters, maybe we need to specifyc that they should be url encoded ?

Reply:
Great point @chmouel. 
I will add it.

Reply:
added the point that the url must be url-encoded.